### PR TITLE
added an 'open' prop on <Drawer>

### PR DESCRIPTION
**Pull request by wakooka**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,6 @@
 # Unreleased
 
+- **[UPDATE]** Added an 'open' prop on `<Drawer>`
```
Comment (Bidjii) [inline CHANGELOG.md]: `'open'` > `` `open` `` if you were looking for the inline code style

File: src/drawer/index.tsx [hunk 2]
```diff
@@ -30,30 +28,23 @@ export default class Drawer extends PureComponent <DrawerProps, DrawerState> {
...
-    // stuff that changed in the same animation frame.
-    requestAnimationFrame(() => {
-      // This nested requestAnimationFrame here is only needed for Firefox
-      requestAnimationFrame(this.open)
-    })
   }
 
-  componentWillReceiveProps({ width }: DrawerProps) {
+  componentDidUpdate({ width, open }: DrawerProps) {
     if (this.props.width !== width) {
```
Comment (loick) [inline src/drawer/index.tsx]: It won't work anymore. This was done before the next render. Here, if you don't have any render after the componentDidUpdate, you won't see any update of the this.contentStyles.width.

File: src/drawer/index.tsx [hunk 3]
```diff
@@ -105,34 +83,36 @@ export default class Drawer extends PureComponent <DrawerProps, DrawerState> {
 
   scrollLock = () => {
     if (canUseDOM) {
-      document.documentElement.classList.add('scroll-lock')
+      document.documentElement.classList.add('kirk-scroll-lock')
```
Comment (loick) [inline src/drawer/index.tsx]: We are using a polyfill on the SPA, classList doesn't seem to be supported by our audience

File: src/drawer/index.unit.tsx [hunk 4]
```diff
@@ -8,47 +8,48 @@ const defaultProps = {
...
+      </Drawer>,
+    )
     expect(wrapper.hasClass(customClassName)).toBe(true)
   })
 
   it('Should call `onOpen` and `scrollLock` after open', () => {
     const onOpen = jest.fn()
-    const onChange = jest.fn()
+    const onTransitionEnd = jest.fn()
     const wrapper = mount(
```
Comment (loick) [inline src/drawer/index.unit.tsx]: Remove the usage of mount, use shallow instead

File: src/drawer/index.unit.tsx [hunk 5]
```diff
@@ -8,47 +8,48 @@ const defaultProps = {
...
         body
       </Drawer>,
     )
     const scrollLock = jest.spyOn(wrapper.instance(), 'scrollLock')
-    wrapper.instance().open()
-    wrapper.find('.scrollableContent').simulate('transitionEnd')
+    expect(onOpen).toHaveBeenCalledTimes(0)
+    expect(scrollLock).toHaveBeenCalledTimes(0)
+    wrapper.setProps({ open: true })
+    wrapper.find('.kirk-drawer-scrollableContent').simulate('transitionEnd')
```
Comment (loick) [inline src/drawer/index.unit.tsx]: Avoid using simulate and call the instance if necessary

File: src/drawer/index.tsx (hunk 2 above)
Comment (wakooka) [reply src/drawer/index.tsx]: Moved to `render`

File: src/drawer/index.tsx (hunk 3 above)
Comment (wakooka) [reply src/drawer/index.tsx]: as discussed IRL, classList seems well supported.

File: src/drawer/index.unit.tsx (hunk 4 above)
Comment (wakooka) [reply src/drawer/index.unit.tsx]: done

File: src/drawer/index.unit.tsx (hunk 5 above)
Comment (wakooka) [reply src/drawer/index.unit.tsx]: Done

File: src/drawer/story.tsx [hunk 6]
```diff
@@ -84,38 +84,39 @@ class DrawerDemo extends Component {
...
   }
 
-  onToggleDrawer = (open: boolean) => {
-    this.setState({ isOpen: open })
-  }
-
   toggleDrawer = () => {
-    if (this.drawer) {
-      this.drawer.close()
+    if (this.state.isOpen) {
```
Comment (pith) [inline src/drawer/story.tsx]: nit: you could just do a toggle

```
toggleDrawer = () => {
  this.setState(({ isOpen })=> ({ isOpen: !isOpen }))
}
```

Comment (wakooka) [reply src/drawer/story.tsx]: I had a typescript issue and I didn't really want to spend time investigating on this, so I merged it  :/

**Pull request merged**